core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java

```diff
@@ -0,0 +1,213 @@
+ * {@link ServerBuilder#annotatedService()}.
+ *
+ * <p>Call {@link #buildAnnotated(Object)} to build the {@link Service} and return to the {@link ServerBuilder}.
+ *
+ * <pre>{@code
+ * ServerBuilder sb = Server.builder();
+ * sb.annotatedService()                                    // Returns an instance of this class
+ *   .requestTimeout(5000)
+ *   .maxRequestLength(8192)
+ *   .exceptionHandler((ctx, request, cause) -> HttpResponse.of(400))
+ *   .pathPrefix("/foo")
+ *   .verboseResponses(true)
+ *   .contentPreview(500)
+ *   .buildAnnotated(new Service())                         // Return to the ServerBuilder.
+ *   .build();
+ * }</pre>
+ *
+ * @see ServiceBindingBuilder
+ */
+public final class AnnotatedServiceBindingBuilder extends AbstractServiceConfigSetters {
+
+    private final ServerBuilder serverBuilder;
+    private final Builder<ExceptionHandlerFunction> exceptionHandlerFunctionBuilder = ImmutableList.builder();
+    private final Builder<RequestConverterFunction> requestConverterFunctionBuilder = ImmutableList.builder();
+    private final Builder<ResponseConverterFunction> responseConverterFunctionBuilder = ImmutableList.builder();
+    private String pathPrefix = "/";
+
+    AnnotatedServiceBindingBuilder(ServerBuilder serverBuilder) {
+        this.serverBuilder = requireNonNull(serverBuilder, "serverBuilder");
+    }
+
+    @Override
+    void serviceConfigBuilder(ServiceConfigBuilder serviceConfigBuilder) {
+        serverBuilder.serviceConfigBuilder(serviceConfigBuilder);
+    }
+
+    /**
+     * Sets the path prefix to be used for this {@link AnnotatedServiceBindingBuilder}.
+     * @param pathPrefix string representing the path prefix.
+     */
+    public AnnotatedServiceBindingBuilder pathPrefix(String pathPrefix) {
+        this.pathPrefix = requireNonNull(pathPrefix, "pathPrefix");
+        return this;
+    }
+
+    /**
+     * Adds the given {@link ExceptionHandlerFunction} to this {@link AnnotatedServiceBindingBuilder}.
+     */
+    public AnnotatedServiceBindingBuilder exceptionHandler(ExceptionHandlerFunction exceptionHandlerFunction) {
+        requireNonNull(exceptionHandlerFunction, "exceptionHandler");
+        exceptionHandlerFunctionBuilder.add(exceptionHandlerFunction);
+        return this;
+    }
+
+    /**
+     * Adds the given {@link ResponseConverterFunction} to this {@link AnnotatedServiceBindingBuilder}.
+     */
+    public AnnotatedServiceBindingBuilder responseHandler(ResponseConverterFunction responseConverterFunction) {
+        requireNonNull(responseConverterFunction, "responseConverterFunction");
+        responseConverterFunctionBuilder.add(responseConverterFunction);
+        return this;
+    }
+
+    /**
+     * Adds the given {@link RequestConverterFunction} to this {@link AnnotatedServiceBindingBuilder}.
+     */
+    public AnnotatedServiceBindingBuilder requestHandler(RequestConverterFunction requestConverterFunction) {
```

Comment:
We use the word `converter` for the request and response. Could you rename it? `requestConverter`